packages/emitter-framework/src/python/components/class-declaration/class-declaration.tsx

```diff
@@ -174,10 +184,48 @@ export function ClassDeclaration(props: ClassDeclarationProps) {
   const { $ } = useTsp();
 
   // If we are explicitly overriding the class as abstract or the type is not a model, we need to create an abstract class
-  let abstract =
+  const abstract =
     ("abstract" in props && props.abstract) || ("type" in props && !$.model.is(props.type));
-  let docElement = createDocElement($, props);
-  let basesType = createBasesType($, props, abstract);
+  const docElement = createDocElement($, props);
+
+  const extraBases = [];
+  let typeVars = null;
+  const typeArgs = [];
+  if (isTypedClassDeclarationProps(props)) {
+    if (
+      !props.type.isFinished &&
+      (props.type.node as TemplateDeclarationNode)?.templateParameters
+    ) {
```

Comment:
@mauriciogardini this is what I was referring to here https://github.com/pinterest/typespec/pull/45#discussion_r2417494357